### PR TITLE
Dispose of ITextDocument(s) when closing preview view

### PR DESCRIPTION
**Pull request by dibarbet**
Resolves https://github.com/dotnet/roslyn/issues/75492

Debugged and verified that the language client now unhooks from these buffers.

### CODE REVIEW

File: src/EditorFeatures/Core/Preview/AbstractPreviewFactoryService.cs [hunk 1]
```diff
@@ -709,6 +711,15 @@ private async ValueTask<IDifferenceViewerPreview<TDifferenceViewer>> CreateNewDi
 
             rightWorkspace?.Dispose();
             rightWorkspace = null;
+
+            var inertContentType = _contentTypeRegistryService.GetContentType(StandardContentTypeNames.Inert);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Preview/AbstractPreviewFactoryService.cs:715]: we should doc why we're doing this.

File: src/EditorFeatures/Core/Preview/AbstractPreviewFactoryService.cs [hunk 2]
```diff
@@ -709,6 +711,15 @@ private async ValueTask<IDifferenceViewerPreview<TDifferenceViewer>> CreateNewDi
...
+            buffersToClose = buffersToClose
+                .Add(originalBuffer)
+                .Add(changedBuffer);
+            foreach (var buffer in buffersToClose)
+            {
+                buffer.ChangeContentType(inertContentType, null);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Preview/AbstractPreviewFactoryService.cs:721]: hamed parameter for null.

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/EditorFeatures/Core/Preview/AbstractPreviewFactoryService.cs [hunk 3]
```diff
@@ -709,6 +711,17 @@ private async ValueTask<IDifferenceViewerPreview<TDifferenceViewer>> CreateNewDi
 
             rightWorkspace?.Dispose();
             rightWorkspace = null;
+
+            // We're closing - change the content type of the buffers we created to inert
+            // as a signal for the editor / LSP client to unhook from them.
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Preview/AbstractPreviewFactoryService.cs:716]: They're not using the view disconnecting to figure that out?

File: src/EditorFeatures/Core/Preview/AbstractPreviewFactoryService.cs [hunk 4]
```diff
@@ -709,6 +711,17 @@ private async ValueTask<IDifferenceViewerPreview<TDifferenceViewer>> CreateNewDi
...
+            var inertContentType = _contentTypeRegistryService.GetContentType(StandardContentTypeNames.Inert);
+            buffersToClose = buffersToClose
+                .Add(originalBuffer)
```
Comment (ToddGrun) [inline src/EditorFeatures/Core/Preview/AbstractPreviewFactoryService.cs:719]: >originalBuffer [](http://example.com/codeflow?start=21&length=14)

Would it make sense for the caller to specify all the buffers to close? We take in a param that indicates what buffers to close, seems like we should just have the caller specify all the buffers to close in it.

**Comment by dibarbet** [comment]
After talking with Jason/Kayle offline - we decided to go a different route.  Now we instead just dispose of the ITextDocument we create (it is IDisposable already, but we weren't disposing).  There is a corresponding change on the LSP client side to ensure the diagnostics state is removed - https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient/pullrequest/586188

**Comment by jasonmalinowski** [comment]
@dibarbet Don't forget to change the PR title.

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**